x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -773,35 +774,53 @@ private static void checkRemoteEnrich(LogicalPlan plan, Set<Failure> failures) {
      * @param typeNameProvider provider for the type name to add in the failure message
      * @param failures         failures collection to add to
      */
-    private static void checkNotPresentInDisjunctions(
+    private static void checkFullTextSearchDisjunctions(
         Expression condition,
         java.util.function.Function<FullTextFunction, String> typeNameProvider,
         Set<Failure> failures
     ) {
         condition.forEachUp(Or.class, or -> {
-            checkNotPresentInDisjunctions(or.left(), or, typeNameProvider, failures);
-            checkNotPresentInDisjunctions(or.right(), or, typeNameProvider, failures);
+            boolean left = checkFullTextSearchInDisjunctions(or.left());
+            boolean right = checkFullTextSearchInDisjunctions(or.right());
+            if (left ^ right) {
+                Holder<String> elementName = new Holder<>();
+                if (right) {
+                    or.left().forEachDown(FullTextFunction.class, ftf -> elementName.set(typeNameProvider.apply(ftf)));
+                } else {
+                    or.right().forEachDown(FullTextFunction.class, ftf -> elementName.set(typeNameProvider.apply(ftf)));
+                }
+                failures.add(
+                    fail(
+                        or,
+                        "Invalid condition [{}]. {} can be used as part of an OR condition, "
+                            + "but only if other full text functions are used as part of the condition",
+                        or.sourceText(),
+                        elementName.get()
+                    )
+                );
+            }
         });
     }
 
     /**
-     * Checks whether a condition contains a disjunction with the specified typeToken. Adds to failure if it does.
+     * Checks whether an expression contains just full text functions or negations (NOT) and combinations (AND, OR) of full text functions
      *
-     * @param parentExpression parent expression to add to the failure message
-     * @param or               disjunction that is being checked
-     * @param failures         failures collection to add to
+     * @param expression parent expression to add to the failure message
```

Comment:
trivially, the javadoc should be updated. It's just the expression to check, right?

Reply:
Yep! Corrected that